quickstarts/microsoft.mobilenetwork/mobilenetwork-update-packet-core-control-plane/main.bicep

```diff
@@ -0,0 +1,31 @@
+@description('Region where the mobile network will be deployed (must match the resource group region)')
+param location string = resourceGroup().location
+
+@description('The name for the existing packet core control plane / site name')
+param existingSiteName string = 'myExampleSite'
+
+@description('The version of packet core to use. Only set this field when instructed to by your support engineer.')
+param version string = ''
+
+// Need to fetch the existing object's state in a separate deployment to workaround circular dependency error
+module fetchedData './nestedtemplates/fetch.bicep' = {
```

Comment:
Yes, we're just trying to update the version. I've got a requirement to allow customers to upgrade their existing packet core version using a template since some customers are only using the quickstart templates to manage their resources.

If we use a simple template which just declares a packet core control plane object, the UX is poor since it requires the user to re-enter all the existing configuration. Therefore, I've written this template which gets those values from the existing object, so the user just needs to enter the desired version.

Does that seem ok to you? Or do you think we need to use the simple template and require the user to fill all the fields?

Reply:
We need to make sure the template will succeed in greenfield or brownfield scenarios - unfortunately the only way to do that is to be completely declarative.  So users would have to submit all the info needed to create it if it does not already exist.  

You can use a uiDefinition file to simplify the portal experience if that's your main use case.

Reply:
Could you explain how the uiDefinition file can simplify the portal experience? I've looked at how it's used for other quickstart templates, but I can't see any example of how it can pull values from existing objects.

Reply:
There is a control that will look up a given resourceType, see: 

https://docs.microsoft.com/en-us/azure/azure-resource-manager/managed-applications/microsoft-solutions-resourceselector

And then a quick sample since I couldn't find any documented:

https://gist.github.com/bmoore-msft/03c561aae8f77218e33199d7b1874833